### PR TITLE
AV-191380 : Code and Documentation changes for AKO Operator 1.11.1 Release

### PR DESCRIPTION
**Pull request by DixitAakash**

### CODE REVIEW

File: ako-operator/controllers/crd.go [hunk 1]
```diff
@@ -295,6 +327,17 @@ func createCRDs(cfg *rest.Config, log logr.Logger) error {
 	return nil
 }
 
+func createK8sObjcetsFromManifests(cfg *rest.Config, ako akov1alpha1.AKOConfig, log logr.Logger) error {
```
Comment (pkoshtavmware) [inline ako-operator/controllers/crd.go:330]: typo in function name

Comment (DixitAakash) [reply ako-operator/controllers/crd.go]: Typo has been corrected

File: ako-operator/controllers/utils.go [hunk 2]
```diff
@@ -251,17 +291,54 @@ func getEnvVars(ako akov1alpha1.AKOConfig, aviSecret v1.Secret) []v1.EnvVar {
...
+	return envVars
+}
+
+func getEnvVarsForGateway(ako akov1alpha1.AKOConfig) []v1.EnvVar {
+	envVars := []v1.EnvVar{}
+	for k, v := range ConfigMapEnvVarsGateway {
 		envVar := v1.EnvVar{
 			Name: k,
 			ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: &v1.SecretKeySelector{
```
Comment (akshayhavile) [inline ako-operator/controllers/utils.go:262]: have we taken out secret as secret shouldn't be present as part of ENV variable? Is AKOO responsible for creating secret objects from values.yaml? How secret will be generated?

File: docs/akoconfig.md [hunk 3]
```diff
@@ -80,13 +82,22 @@ spec:
   pvc: ""
   mountPath: "/log"
   logFile: "avi.log"
+  featureGates:
+    gatewayAPI: true
+  gatewayAPI:
+    image:
+      repository: "projects.registry.vmware.com/ako/ako-gateway-api@sha256:089e4ccd0642954cc031eba541b0e3c84b5d88c1c96ae5f788edeed30ecfd26d"
```
Comment (akshayhavile) [inline docs/akoconfig.md:89]: Can we mention version instead of SHA?

File: ako-operator/controllers/utils.go (hunk 2 above)
Comment (DixitAakash) [reply ako-operator/controllers/utils.go:262]: I have taken out the env variable that was created from the avi secret. The avi secret is not created by AKOO. It has to be created by the user before applying AKOConfig CRD object.

File: docs/akoconfig.md (hunk 3 above)
Comment (DixitAakash) [reply docs/akoconfig.md]: Done

File: ako-operator/controllers/configmap_test.go [hunk 4]
```diff
@@ -56,14 +56,14 @@ var cmJson = `
 		"enableEvents": "true",
 		"logLevel": "INFO",
 		"nsxtT1LR": "",
-		"nodeNetworkList": "[{\"cidrs\":[\"10.10.10.0/24\"],\"networkName\":\"test-nw\"}]",
+		"nodeNetworkList": "[{\"cidrs\":[\"10.10.10.0/24\"],\"networkName\":\"test-nw\",\"networkUUID\":\"\"}]",
```
[the remark this answers is not in the document]
Comment (DixitAakash) [reply ako-operator/controllers/configmap_test.go]: I have added test for updating node network list and vip network list in cm with **networkUUID** in place of **networkName**. Have kept **networkName** as default. Also tested ako working with networkUUID specified in akoconfig.

**Review by arihantg — APPROVED** [review]

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**